packages/react-native/Libraries/Components/TextInput/TextInput.js

```diff
@@ -1206,6 +1207,28 @@ function InternalTextInput(props: Props): React.Node {
     }
   }, [inputRef]);
 
+  useLayoutEffect(() => {
+    if (
+      inputRef.current != null &&
+      prevSecureTextEntry.current !== props.secureTextEntry
+    ) {
+      viewCommands.setTextAndSelection(
+        inputRef.current,
+        mostRecentEventCount,
+        lastNativeText,
```

Comment:
Hmm… Flow is unhappy with this because `lastNativeText` is `?string` (but the argument must be `string`).

How did you decide between using `lastNativeText` (and `lastNativeSelection`) instead of `text` (and `selection`)?

Given that this layout effect is invoked after the one online 1152, it seems like we should actually maybe using the latter. Thoughts?

Reply:
Perhaps you're right, my idea was just to preserve the actual state based on the `lastNativeText` and `lastNativeSelection`, which are internal while `text` and `selection` are external and are optional.

Reply:
The more I think about it, the more I wonder whether this logic should actually be incorporated into the layout effect on line 1151. If `props.secureTextEntry` and `text` both change in the same update, we would end up invoking `viewCommands.setTextAndSelection` twice (and as currently written, with different values — once with `text` and once with `lastNativeText`).

What do you think about merging this into that layout effect by doing the following?

* Include `props.secureTextEntry` into the earlier layout effect's dependencies array.
* Adding the `prevSecureTextEntry.current !== props.secureTextEntry` check to line 1169.

This proposed change would also eliminate the question of whether we should be calling with `text` or `lastNativeText`.

Reply:
@yungsters 

Sorry for the delay, I got caught in some work 😅 

I've been doing some thinking and I'm not sure if what you're suggesting will do it because what I want to achieve is to enforce the previous component state into the native component.

I'm not sure how TextInput should behave in these situations, but what makes sense to me is:

1. Native component gets updated
2. React component receives native state update and checks if it differs from props (controlled component)
2.1. If there are no text and/or selection props, use react component state instead 
2.2. If it differs, update react component state and enforce props into native state
2.3. If it's equal, then don't do anything

This way, in controlled and uncontrolled situations, the native component should always reflect either the props or the react component state.

Let me know what you think 😄

Reply:
@yungsters bump